src/DAT/xmap.jl

```diff
@@ -406,13 +406,20 @@ compute_to_zarr(ds, "output.zarr")
 function xmap end
 
 import Base.mapslices
-function mapslices(f, d::YAXArray, addargs...; dims, kwargs...)
+function mapslices(f, d::YAXArray, addargs...; dims, dropdims=false, kwargs...)
```

Comment:
@briochemc by default the new mapslices in YAXArrays will not drop singleton dimensions anymore, which is in line with what Base and DimensionalData mapslices are doing and, more importantly makes stacking multiple operations along the same axis more efficient. I have added the `dropdims` keyword to mapslices which restores the old behavior when set to `true`, I hope this would work for you.

Reply:
Sounds good to me!